src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -861,10 +860,21 @@ static int getReadPositionAtReferencePositionIgnoreSoftClips(final SAMRecord rec
         rec.setCigar(newCigar);
         // Since the read effectively got shifted forward by turning the clips into matches, the query position needs
         // also to be moved forward bye posShift so that it's still querying the same base.
-        readPosition = SAMRecord.getReadPositionAtReferencePosition(rec, pos + posShift, false);
+        final int readPosition = SAMRecord.getReadPositionAtReferencePosition(rec, pos + posShift, true);
+
         rec.setCigar(oldCigar);
 
-        return readPosition;
+        // if this returns zero, it means that there's a deletion at the position of the desired base,
+        // if the read is on the positive strand readPosition should be incremented by one to get the
+        // base __following__ the deletion rather than the one preceding it (which is what returnLastBaseIfDeleted argument in
+        // getReadPositionAtReferencePosition will do)
+        final int readPositionZeroOnDeletion = SAMRecord.getReadPositionAtReferencePosition(rec, pos + posShift, false);
+        final boolean increamentReadPosition = readPositionZeroOnDeletion == 0 &&
+                !rec.getReadNegativeStrandFlag() &&
+                readPosition < rec.getReadLength() - 1;
```

Comment:
this is to protect from a stupidly aligned read that has a deletion at the end somehow, so there are no bases __following__ that base.